### PR TITLE
Add Mock.Of<T> and MockRepository support to ConstructorArgumentsShouldMatchAnalyzer

### LINKED ISSUE #122: Extend analyzers to cover MockRepository
Moq also provides a MockRepository to share common setup between mocks. https://github.com/devlooped/moq/blob/main/src/Moq/MockRepository.cs

We should extend the appropriate analyzers to cover them as well.

### PR DESCRIPTION
**Pull request by rjmurillo**
The majority of this change is mechanical (moving code from one location to another, replacing magic strings with constants, etc.). There is a functional update that completely changes the `ConstructorArgumentsShouldMatchAnalyzer`:

- consolidating two constructor analyzers now into a single shared source
- improving the performance of the constructor analyzer
- adding support for checking delegate
- adding support for `Mock.Of<T>` and `MockRepository` features of `Moq`

image
_Baseline is an empty analyzer, control is the old interface constructor parameter analyzer, treatment is the new combined analyzer._

Changes

Misc:
- Reduced severity of AV analyzer items (see #139)
- Move magic strings to constants
- Moved much of the common code into a `Common` folder and namespace

Documentation:
- Fixed links to documentation for shipped analyzers
- In analyzer rule help, link to commit id so documentation snaps to the version of the analyzer

Analyzers:
- Moved rule ID to constants file
- Added extension factory methods to help create `Diagnostic`
- Consolidated two constructor parameter analyzers (Moq1001 and Moq1002) into a single analyzer with 30% better performance
- `ConstructorArgumentsShouldMatchAnalyzer` now supports `Mock.Of<T>` syntax and `MockRepository` patterns

Benchmarks:
- Add the ability to have a control and treatment as well as "baseline", allowing A/B performance benchmarking of implementations
- Added benchmark for `ConstructorArgumentsShouldMatchAnalyzer`

Test:
- Added extension methods to include all Moq versions under test or just the latest
- Organized the `ConstructorArgumentsShouldMatchAnalyzer` test collateral (it's the highest volume) into partial files
- Added Live Unit Test configuration when running in Visual Studio
- Added code coverage filter for benchmark code if using ReSharper or Rider

_Note: while code coverage is improved, it's still below 95%. This is due to some guard cases that need coverage. For example, #141_

Resolves #122
For Moq1001 and Moq1002, #85

### CODE REVIEW

File: src/Moq.Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs [hunk 1]
```diff
@@ -1,252 +1,487 @@
...
-            Diagnostic diagnostic = objectCreation.ArgumentList.CreateDiagnostic(Rule);
-            context.ReportDiagnostic(diagnostic);
+            return;
         }
+
+        // Full check
+        SymbolInfo symbolInfo = context.SemanticModel.GetSymbolInfo(newExpression, context.CancellationToken);
+
+        if (symbolInfo.Symbol is not IMethodSymbol mockConstructorMethod
+            || mockConstructorMethod.MethodKind != MethodKind.Constructor
```
Comment (MattKotsenas) [inline src/Moq.Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs:262]: nit: Can use your `IsConstructor` extension method here

**Review by MattKotsenas — DISMISSED** [review]

File: src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    /// <param name="compilationOptions">The compilation options that will be used to determine if the diagnostic is currently suppressed.</param>
+    /// <param name="descriptor">The diagnostic descriptor to check.</param>
+    /// <returns>True if the diagnostic is currently suppressed.</returns>
+    internal static bool IsAnalyzerSuppressed(this CompilationOptions compilationOptions, DiagnosticDescriptor descriptor)
+    {
+        switch (descriptor.GetEffectiveSeverity(compilationOptions))
+        {
+            case ReportDiagnostic.Suppress:
+                return true;
+            case ReportDiagnostic.Default:
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs:17]: This I think can be removed. I was tripping over one of those AV analyzers which got removed in #145.

File: Moq.Analyzers.sln.DotSettings [hunk 3]
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply Moq.Analyzers.sln.DotSettings:1]: Yes. Foo.sln.DotSettings is a file that ReSharper uses to save "team shared" settings. ReSharper has settings and configuration abilities that are specific to ReSharper and do not have an equivalent `.EditorConfig` standard.

File: src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+internal static class CompilationOptionsExtensions
+{
+    /// <summary>
+    /// Determines if the diagnostic identified by the given identifier is currently suppressed.
+    /// </summary>
+    /// <param name="compilationOptions">The compilation options that will be used to determine if the diagnostic is currently suppressed.</param>
+    /// <param name="descriptor">The diagnostic descriptor to check.</param>
+    /// <returns>True if the diagnostic is currently suppressed.</returns>
+    internal static bool IsAnalyzerSuppressed(this CompilationOptions compilationOptions, DiagnosticDescriptor descriptor)
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs:11]: Follow up tracked in #147

File: src/Moq.Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs (hunk 1 above)
Comment (rjmurillo) [reply src/Moq.Analyzers/ConstructorArgumentsShouldMatchAnalyzer.cs:262]: The test on L263 requires the extracted ctor symbol. I want to clean up a bunch of redundancies in the code like this. I'll take a pass on another PR

File: src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rjmurillo) [reply src/Moq.Analyzers/Common/CompilationOptionsExtensions.cs]: Can be. Removed

**Pull request merged**